src/Document/VariantDocument.php

```diff
@@ -61,6 +75,22 @@ public function __construct()
     }
 
     /**
+     * @return int
+     */
+    public function getId(): int
+    {
+        return $this->id;
+    }
+
+    /**
+     * @param int $id
+     */
+    public function setId(int $id): void
```

Comment:
If it's Sylius' id, it can be anything, so mixed (and falling back to string) is the better choice.

Reply:
That is Doctrine's id, the `code` has it's own `code` field in the documents